presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java

```diff
@@ -378,6 +378,18 @@ public void testReduceAgg()
         assertQuery("SELECT orderkey, array_sort(reduce_agg(linenumber, array[], (s, x) -> s || x, (s, s2) -> s || s2)) FROM lineitem GROUP BY orderkey");
     }
 
+    @Test
+    public void testNaNValueAgg()
+    {
+        // Fix Velox to get these tests passed.
+        String notEqualRowsErrorMsg = "*.not equal.*";
+        assertQueryError("SELECT DISTINCT a/a FROM (VALUES (0.0e0), (0.0e0)) x (a)", notEqualRowsErrorMsg);
+        assertQueryError("SELECT * FROM (VALUES nan(), nan(), nan()) GROUP BY 1", notEqualRowsErrorMsg);
+        assertQueryError("SELECT a, b, c FROM (VALUES ROW(nan(), 1, 2), ROW(nan(), 1, 2)) t(a, b, c) GROUP BY 1, 2, 3", notEqualRowsErrorMsg);
```

Comment:
Its interesting this didn't work either. Is this because of GROUP BY 1, 2, 3 ? If we do just GROUP BY 1 does this work ?

Reply:
The values clause is turned into a three column table of normalized (simple) values. So you get the failure in the equals for the NaN.

When changed to GROUP BY 1 the query fails with 
```
Runtime line 1:11: 'b' must be an aggregate expression or appear in GROUP BY clause
```
because the other columns in the projection have no aggregation defined either via group by, or sum etc.
Running 
```
SELECT a, SUM(b), SUM(c) FROM (VALUES ROW(nan(), 1, 2), ROW(nan(), 1, 2)) t(a, b, c) GROUP BY 1
```
results in the error as expected.

I tried various things to have the ROW as values in a single column but it somehow always either results in two columns or three columns.

Technically, this should have worked based on the syntax. 
```
SELECT a FROM (VALUES (ROW(nan(), 1, 2)), (ROW(nan(), 1, 2)) ) t(a) GROUP BY 1
```
Maybe I'm missing something. This complains that the table has 3 columns.

Reply:
@czentgr : I see. I expected this to be a single column with ROW types as well. I guess right now the ROW usage is redundant.